doc/md/guides/overview/modern-database-ci-cd.mdx

```diff
@@ -0,0 +1,233 @@
+slug: modern-database-ci-cd
+---
+
+![](https://atlasgo.io/uploads/ci-cd-guide/database-ci-cd-workflow.png)
+
+Continuous integration and continuous delivery (CI/CD) are tenets of modern software development.
+CI/CD enables teams to deliver software faster, with higher quality, and with less risk. However,
+CI/CD is not just for application code.  Teams can also apply CI/CD principles to their databases,
+to extend the benefits of CI/CD to one of the most critical components of their stack.
+
+This document describes the recommended workflow for teams using Atlas to achieve continuous integration
+and continuous delivery (CI/CD) for their databases.
+
+## Principles
+
+1. **Changes to the database are planned automatically**. Given the desired state of the database,
+   the system should automatically generate a plan for how to get from the current state to the desired
+   state.
+2. **Changes to the database schema are stored in a versioned migration directory.** All planned changes
+   to the database are checked in to a versioned migration directory.  This directory contains SQL scripts,
+   which are executed in order to apply the changes to the database.
+3. **Changes to the database are validated during CI.**  All changes to the database are tested and evaluated
+   against a set of governing policies.
+4. **Changes to the database are deployed via automation.**  No manual steps are required to deploy changes to the
+   database.  All changes are deployed via a CI/CD pipeline.
+
+## Workflow
+
+This section follows a single change to your database schema from the time it is planned to the time it is deployed.
+The purpose of this section is not to provide extensive instructions for how to configure your CI/CD pipeline, but
+rather to provide a high-level overview of the recommended workflow.
+
+### 1. Create a new branch
+
+Create a new branch to track this change to the database.
+
+Example:
+
+```
+git checkout -b add-new-index
+```
+
+### 2. Change the desired state of the database
+
+Similar to modern Infrastructure-as-Code tools, Atlas uses a declarative approach to database management.
+This means that you do not need to write SQL scripts to apply changes to the database.  Instead, you simply
+declare the desired state of the database, and Atlas will automatically generate the SQL scripts required
+to apply the change.
+
+In Atlas, the desired state of the database can be provided in many forms. For example, you can provide
+it as a plain SQL file, using [Atlas HCL](/atlas-schema/hcl), as a reference to your [ORM](/atlas-schema/external),
+or even as a connection to an existing database.  For the purposes of this example, we will use a plain SQL file.
+
+Example:
+
+```diff {title="schema.sql"}
+CREATE TABLE users (
+  id INT AUTO_INCREMENT PRIMARY KEY,
+  name VARCHAR(255) NOT NULL,
+  email VARCHAR(255) NOT NULL,
+  created_at TIMESTAMP NOT NULL DEFAULT CURRENT_TIMESTAMP,
++  INDEX idx_users_email (email)
+);
+```
```

Comment:
While this is high-level, maybe we should still provide "ready-to-copy" snippets to follow along?